include/boost/uuid/detail/random_provider.hpp

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2017 James E. King III
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//   http://www.boost.org/LICENCE_1_0.txt)
+//
+// Platform-specific random entropy provider
+//
+
+#ifndef BOOST_UUID_DETAIL_RANDOM_PROVIDER_HPP
+#define BOOST_UUID_DETAIL_RANDOM_PROVIDER_HPP
+
+#include <boost/core/noncopyable.hpp>
+#include <boost/cstdint.hpp>
+#include <boost/limits.hpp>
+#include <boost/static_assert.hpp>
+#include <boost/uuid/entropy_error.hpp>
+
+#include <boost/uuid/detail/random_provider_detect_platform.hpp>
+#include BOOST_UUID_RANDOM_PROVIDER_BASE_INCLUDE()
+
+namespace boost {
+namespace uuids {
+namespace detail {
+
+//! \brief Contains code common to all random_provider implementations.
+//! \note  random_provider_base is required to provide this method:
+//!        void get_random_bytes(void *buf, size_t siz);
+//! \note  noncopyable because of some base implementations so
+//!        this makes it uniform across platforms to avoid any  
+//!        porting surprises
+class random_provider
+    : public detail::random_provider_base,
+      public noncopyable
+{
+public:
+    //! Leverage the provider as a SeedSeq for
+    //! PseudoRandomNumberGeneration seeing.
+    //! \note: See Boost.Random documentation for more details
+    template<class Iter>
+    void generate(Iter first, Iter last)
+    {
+        BOOST_STATIC_ASSERT(sizeof(Iter) >= sizeof(unsigned int));
```

Comment:
This checks the iterator size, which is irrelevant. It should check

- whether the `value_type` of the iterator is integral;
- whether the `value_type` of the iterator is unsigned;
- whether `numeric_limits<value_type>::max()` is >= 2^32-1, but this can only be tested at compile time on C++11, so
- whether `sizeof(value_type) * CHAR_BIT >= 32`.

Reply:
Okay, I will update this.

Reply:
I tried this but it didn't work - I'll keep on it.
```
        typedef Iter::value_type iter_value_type;
        BOOST_STATIC_ASSERT(is_integral<iter_value_type>::value == true_type);
        BOOST_STATIC_ASSERT(is_unsigned<iter_value_type>::value == true_type);
        BOOST_STATIC_ASSERT(sizeof(iter_value_type) * CHAR_BIT >= 32);
```
In reading [SeedSeq documentation](http://www.boost.org/doc/libs/1_65_1/doc/html/boost_random/reference.html#boost_random.reference.concepts.seed_sequence), Iter can be a class with a value_type type definition of an unsigned integral and an operator*, or it can be a pointer to an unsigned integral, correct?  I'm not certain (yet) how I would test for both.

Reply:
`typedef std::iterator_traits<Iter>::value_type value_type;`

Reply:
Then just `BOOST_STATIC_ASSERT( boost::is_integral<value_type>::value );`, comparing to the type `true_type` doesn't make sense.

Reply:
It will work in C++11 because `true_type` has a `constexpr` conversion to `true` but is still nonsensical. :-)

Reply:
I want to support C++03, will it still work?
I don't think I have ever used std::iterator_traits before.
I'm learning a ton of new tricks, thanks!